modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/components/export-form/openExportFormModal.es.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayForm from '@clayui/form';
+import {ClayIconSpriteContext} from '@clayui/icon';
+import ClayModal, {useModal} from '@clayui/modal';
+import {render} from '@liferay/frontend-js-react-web';
+import React, {useState} from 'react';
+import {unmountComponentAtNode} from 'react-dom';
+
+import {ExportFormModalBody} from './ExportFormModalBody.es';
+
+export function ExportFormModal({
+	csvExport,
+	exportFormInstanceURL,
+	fileExtensions,
+	portletNamespace,
+}) {
+	const [visible, setVisible] = useState(true);
+
+	const {observer, onClose} = useModal({
+		onClose: () => setVisible(false),
+	});
+
+	return (
+		<>
+			{visible && (
+				<ClayModal observer={observer}>
+					<ClayModal.Header>
+						{Liferay.Language.get('export')}
+					</ClayModal.Header>
+
+					<ClayForm action={exportFormInstanceURL} method="post">
+						<ClayModal.Body>
+							<ExportFormModalBody
+								csvExport={csvExport}
+								fileExtensions={fileExtensions}
+								portletNamespace={portletNamespace}
+							/>
+						</ClayModal.Body>
+
+						<ClayModal.Footer
+							last={
+								<ClayButton.Group spaced>
+									<ClayButton
+										displayType="secondary"
+										onClick={onClose}
+									>
+										{Liferay.Language.get('cancel')}
+									</ClayButton>
+
+									<ClayButton
+										displayType="primary"
+										onClick={onClose}
+										type="submit"
+									>
+										{Liferay.Language.get('ok')}
+									</ClayButton>
+								</ClayButton.Group>
+							}
+						/>
+					</ClayForm>
+				</ClayModal>
+			)}
+		</>
+	);
+}
+
+let container;
+
+export function openExportFormModal({spritemap, ...data}) {
```

Comment:
Hey, @miltonmc. Could you please take a look at this logic to open a modal? Do you see any way other than manipulating the DOM's elements?

Reply:
Just to add my cents, I tried to change this code, which was a copy and paste from [openShareFormModal.es.js](https://github.com/liferay/liferay-portal/blob/e86c4bf65d0d028ee91f7ef228687cd54507405b/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/components/share-form/openShareFormModal.es.js#L102-L128), but the modal was not showing up. I'm open to suggestions 😄